src/main/java/build/buildfarm/common/config/Server.java

```diff
@@ -38,10 +41,22 @@ public enum INSTANCE_TYPE {
   private String publicName;
 
   public String getPublicName() {
+    // use environment override (useful for containerized deployment)
+    if (!Strings.isNullOrEmpty(System.getenv("INSTANCE_NAME"))) {
+      return System.getenv("INSTANCE_NAME");
+    }
+
+    // use configured value
     if (!Strings.isNullOrEmpty(publicName)) {
       return publicName;
-    } else {
-      return System.getenv("INSTANCE_NAME");
+    }
+
+    // derive a value
+    try {
+      return InetAddress.getLocalHost().getHostAddress() + ":" + port;
```

Comment:
Have you tested this with k8s? May throw a comment in here that it needs to be tested, if it hasn't as I think the pod IP could be different from the container IP.

Reply:
good suggestion.  I've now tried this with the local repo k8s files, and to my surprise, it worked.

When looking at the pod I see:
```
kubectl get pods -o wide | grep worker | tr -s " "
shard-worker-56b6cd6d6d-2g96n 1/1 Running 0 73m 10.1.110.85 desktop <none> <none>
```

Which is indeed a different IP than what I see for the service:
```
kubectl get service --all-namespaces | grep worker | tr -s " "
default shard-worker-service ClusterIP 10.152.183.219 <none> 8981/TCP,9091/TCP 74m
```

However, when looking at the endpoint, I see the same pod IP:
```
kubectl get endpoints | grep worker | tr -s " "
shard-worker-service 10.1.110.85:8981,10.1.110.85:9091 74m
```

All this is to say, leaving out `publicName` resulted in `10.1.110.85:8981` being derived which did work in my local k8s deployment and allow me to perform remote bazel builds.  I don't know enough about k8s to claim that this will work for all other deployments (such as yours).  I'll have more experience on that when I do a production level k8s deployment and try to leverage this.

I can imagine a future adjustment where we detect that were inside a k8s pod, and query for the correct publicly visible endpoint or something.  For now, `getLocalHost()` seems fairly robust.